test/tests/selftest/pbs_pbstestsuite.py

```diff
@@ -0,0 +1,222 @@
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.selftest import *
+from ptl.utils.pbs_snaputils import *
+
+
+class TestPBSTestSuite(TestSelf):
+    """
+    Contains tests for pbs_testsuite module's functionality
+    """
+
+    def test_revert_pbsconf_onehost(self):
+        """
+        Test the functionality of PBSTestSuite.revert_pbsconf()
+        for a single host type 1 installation
+        """
+        pbs_conf_filepath = self.du.get_pbs_conf_file(self.server.hostname)
+        pbs_conf_val = self.du.parse_pbs_config(self.server.hostname,
+                                                pbs_conf_filepath)
+        if pbs_conf_val is None or len(pbs_conf_val) < 1:
+            raise ValueError("Could not parse pbs.conf on host %s" %
+                             (server.hostname))
+
+        # Since the setUp already ran, check that the start bits are turned on
+        self.assertEqual(pbs_conf_val["PBS_START_MOM"], "1")
+        self.assertEqual(pbs_conf_val["PBS_START_SERVER"], "1")
+        self.assertEqual(pbs_conf_val["PBS_START_SCHED"], "1")
+        self.assertEqual(pbs_conf_val["PBS_START_COMM"], "1")
+
+        self.server.pi.stop()
+
+        # Now, change pbs.conf to turn the sched off
+        pbs_conf_val["PBS_START_SCHED"] = "0"
+        self.du.set_pbs_config(fin=pbs_conf_filepath, confs=pbs_conf_val)
+
+        # Start PBS again
+        self.server.pi.start()
+
+        # Verify that the scheduler didn't come up
+        self.assertFalse(self.scheduler.isUp())
+
+        # Now call revert_pbsconf() & restart PBS again
+        self.revert_pbsconf()
+
+        # Verify that the scheduler came up and start bit is 1
+        self.assertTrue(self.scheduler.isUp())
+        pbs_conf_val = self.du.parse_pbs_config(self.server.hostname)
+        self.assertEqual(pbs_conf_val["PBS_START_SCHED"], "1")
+
+    def test_revert_pbsconf_remotemom(self):
+        """
+        Test the functionality of PBSTestSuite.revert_pbsconf()
+        with a remote mom setup
+        """
+        remotemom = None
+        for mom in self.moms:
+            if not self.du.is_localhost(mom):
+                remotemom = mom
+                break
+        if remotemom is None:
+            self.skip_test("Test needs at least one remote Mom host,"
+                           " use -p moms=<hostname>")
+
+        pbs_conf_val = self.du.parse_pbs_config(self.server.hostname)
+        if pbs_conf_val is None or len(pbs_conf_val) < 1:
+            raise ValueError("Could not parse pbs.conf on host %s" %
+                             (server.hostname))
+
+        # Since the test requested for a mom, the server host's mom should not
+        # be ON, the rest should be set to 1
```

Comment:
Why does it matter if the server's mom is up or not?  We're checking a remote mom.